### ISSUE TITLE
Try a gcc 6 build in travis-ci

### ISSUE THREAD

**Issue by isuruf** [symengine/symengine#1134]
`gcc-6` package is available in travis.

**Comment by ShikharJ** [comment]
@isuruf I'd like to take this up as well, if it is fine.

**Comment by ShikharJ** [comment]
The result that I got: https://travis-ci.org/ShikharJ/symengine/jobs/179200101
Using the code:
```
export CC=gcc-6
export CXX=g++-6
export GCOV_EXECUTABLE=gcov-6
```

**Comment by isuruf** [comment]
Yes, there are new warnings that should be fixed.

**Comment by ShikharJ** [comment]
Would you like me to work on them?

**Comment by isuruf** [comment]
Yes, let us know if you have any questions.

**Comment by isuruf** [comment]
Closed via #1142
